server/src/main/scala/org/http4s/server/middleware/HSTS.scala

```diff
@@ -0,0 +1,27 @@
+package org.http4s
+package server
+package middleware
+
+import org.http4s.headers.`Strict-Transport-Security`
+import fs2._
+import fs2.interop.cats._
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+
+/** [[Middleware]] to add HTTP Strict Transport Security (HSTS) support adding
+  * the Strict Transport Security headers
+  */
+object HSTS {
+
+  def apply(service: HttpService, maxAge: FiniteDuration = 365.days, includeSubDomains: Boolean = true, preload: Boolean = false): HttpService = Service.lift { req =>
+
+    val header = `Strict-Transport-Security`.unsafeFromDuration(maxAge, includeSubDomains, preload)
```

Comment:
Is `delta-seconds` common enough in the spec that we just need a duration type with a granularity of seconds?  I don't like introducing another utility type, but this generic use of Long isn't ideal either.

I think I like your proposed solution here regardless.